### ISSUE TITLE
Example in section "Type checking in the lazy API" does not raise an error

### ISSUE THREAD

**Issue by aglebov** [pola-rs/polars-book#359]
The example provided in the "Type checking in the lazy API" section (https://pola-rs.github.io/polars-book/user-guide/lazy/schemas/#type-checking-in-the-lazy-api) does not seem to raise `SchemaError`:
```
pl.DataFrame({"foo": ["a", "b", "c"], "bar": [0, 1, 2]}).lazy().with_columns(
    pl.col("bar").round(0)
)
```

Even having strings in the `bar` column does not appear to raise an error:
```
pl.DataFrame({"foo": ["a", "b", "c"], "bar": ["a", "b", "c"]}).lazy().with_columns(
    pl.col("bar").round(0)
)
```

When collecting the results, however, the error is raised: `InvalidOperationError: `round` operation not supported for dtype `str``

I've tried these examples on polars versions 0.18.2 and 0.18.3.

**Comment by braaannigan** [comment]
Thanks @aglebov - I've got a fix coming for this

**Comment by braaannigan** [comment]
Fixed in #380
